### ISSUE TITLE
`soup train --resume` on `task: prm` / `moe_lora_routing` says "Resuming from ...", then retrains from step 0 and overwrites the checkpoints

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#1220]
## What happens

For `task: prm` and `task: moe_lora_routing`, `--resume` is resolved and announced, then dropped. The run starts again at step 0, re-executes every step, and rewrites the existing `checkpoint-N` directories as it passes them, replacing the checkpoints the user asked to resume from. Exit code 0, no warning.

Minimal repro on `main` `99d553f5` (CPU, tiny model). A spy counts the training steps each invocation actually executes:

```python
import json
from pathlib import Path

import transformers
from typer.testing import CliRunner

from soup_cli.cli import app

steps = []  # training steps executed per `soup train` invocation
_training_step = transformers.Trainer.training_step


def counting_step(self, *args, **kwargs):
    steps[-1] += 1
    return _training_step(self, *args, **kwargs)


transformers.Trainer.training_step = counting_step

Path("prm.jsonl").write_text("\n".join(json.dumps(
    {"prompt": f"Solve {i}+1.", "completions": [f"{i}+1 is {i + 1}.", "Done."],
     "labels": [True, True]}) for i in range(8)))
Path("chat.jsonl").write_text("\n".join(json.dumps({"messages": [
    {"role": "user", "content": f"q{i}"}, {"role": "assistant", "content": f"a{i}"}]})
    for i in range(8)))
common = ("training: {epochs: 1, batch_size: 2, gradient_accumulation_steps: 1, save_steps: 2, "
          "quantization: none%s}\n")
Path("prm.yaml").write_text(
    "base: hf-internal-testing/tiny-random-LlamaForCausalLM\ntask: prm\n"
    "data: {train: prm.jsonl, format: prm, val_split: 0.0, max_length: 64}\n"
    + common % ", lora: {r: 0}" + "output: ./out_prm\n")
Path("sft.yaml").write_text(
    "base: hf-internal-testing/tiny-random-LlamaForCausalLM\ntask: sft\n"
    "data: {train: chat.jsonl, format: chatml, val_split: 0.0, max_length: 64}\n"
    + common % "" + "output: ./out_sft\n")

runner = CliRunner()
for config in ("prm.yaml", "sft.yaml"):
    for extra in ([], ["--resume", "auto"]):
        steps.append(0)
        res = runner.invoke(app, ["train", "--config", config, "--yes", *extra])
        resumed = [line.strip() for line in res.output.splitlines() if "Resuming from" in line]
        print(config, extra or "(first run)", "| exit", res.exit_code, "|", resumed,
              "| training steps executed:", steps[-1])
```

```
prm.yaml (first run) | exit 0 | [] | training steps executed: 4
prm.yaml ['--resume', 'auto'] | exit 0 | ['Resuming from: out_prm\\checkpoint-4'] | training steps executed: 4
sft.yaml (first run) | exit 0 | [] | training steps executed: 4
sft.yaml ['--resume', 'auto'] | exit 0 | ['Resuming from: out_sft\\checkpoint-4'] | training steps executed: 0
```

A longer run of the same check (the same spy on `Trainer.train`, plus the `trainer_state.json` modification times) adds `task: moe_lora_routing` with two small task adapters:

| task | `--resume auto` prints | argument reaching `Trainer.train()` | steps executed | existing checkpoints rewritten |
|---|---|---|---|---|
| prm | `Resuming from: out_prm\checkpoint-4` | none | 4 of 4 | checkpoint-2, checkpoint-4 |
| moe_lora_routing | `Resuming from: out_mole\checkpoint-4` | none | 4 of 4 | checkpoint-2, checkpoint-4 |
| sft (control) | `Resuming from: out_sft\checkpoint-4` | `out_sft\checkpoint-4` | 0 of 4 | none |

## Why

`PRMTrainerWrapper.train(self, **_kwargs)` (`src/soup_cli/trainer/prm.py:282`) accepts and discards every keyword and calls `self.trainer.train()` with no checkpoint (`prm.py:365`). `MoleRoutingTrainerWrapper` does the same (`src/soup_cli/trainer/mole_routing.py:327`, `:405`). `soup train` resolves `--resume` and prints "Resuming from" for every task (`src/soup_cli/commands/train.py:2268-2285`), then calls `trainer_wrapper.train(..., resume_from_checkpoint=resume_from)` (`train.py:1703-1706`), which these two wrappers ignore. Other wrappers forward it; SFT does so at `src/soup_cli/trainer/sft.py:2234`. The restarted run writes `checkpoint-N` into directories that already exist, so they are overwritten.

The same `**_kwargs` also drops `display`, `tracker` and `run_id`. That half is already known: `_validate_callback_monitoring_task_compat` in `src/soup_cli/config/schema.py` records that these tasks attach no live callback (#802, #1069). This issue is only about resume.

## Fix path

**Wire it.** Both wrappers build a Hugging Face `Trainer` subclass, so the core change is to accept `resume_from_checkpoint` and pass it to `self.trainer.train(...)`, as the other wrappers do. That is enough for PRM, whose checkpoints hold the full model (`model.safetensors`). MoLE needs one more step: its checkpoints contain only the two frozen task adapters (`task_0/`, `task_1/`, 8 tensors each) plus optimizer and scheduler state. The trained gate (`gate.weight`) is written only once, to `mole_gate.pt`, after training ends. Forwarding the checkpoint alone would restore the step counter and optimizer state around a freshly initialised gate, so the gate must be saved into each checkpoint and restored from it.

**Refuse**, until then: `soup train` should exit non-zero on `--resume` / `--hf-resume` for `task: prm` and `task: moe_lora_routing`, with a message naming the task, rather than print "Resuming from" and restart. The resume resolution in `commands/train.py` is the one place to put it, and it runs before any heavy work.

## Acceptance

- [ ] A test trains PRM on the tiny model to `checkpoint-4`, reruns with `--resume auto`, and asserts that 0 training steps run and `trainer.state.global_step` continues from 4. It fails on `main` (4 steps re-run).
- [ ] The same for `moe_lora_routing`, plus: the gate weights after resume equal the gate weights saved at the checkpoint.
- [ ] Variants: `--resume auto`, `--resume <explicit checkpoint dir>`, and a resume from the middle checkpoint (`checkpoint-2`, with 2 steps left to run).
- [ ] No existing checkpoint directory is rewritten by a resumed run (compare `trainer_state.json` modification times).
- [ ] A scan over `src/soup_cli/trainer/*.py` fails when a wrapper's `train()` accepts `resume_from_checkpoint` (by name or through `**kwargs`) without forwarding it, and the task is not refused for `--resume`.
- [ ] If the refusal ships first: a CLI test asserts the non-zero exit and the task name in the message, through an ANSI-strip + whitespace-collapse helper.
